imap_processing/cdf/tests/test_cdf_attribute_manager.py

```diff
@@ -1,58 +1,322 @@
+        == "Solar Physics>Heliospheric Physics"
+    )
+
+    # Testing attributes newly loaded
+    assert cdf_manager.global_attributes["Project"] == "STP>Solar-Terrestrial Physics"
+    assert (
+        cdf_manager.global_attributes["Source_name"]
         == "IMAP>Interstellar Mapping and Acceleration Probe"
     )
+    assert cdf_manager.global_attributes["Mission_group"] == "Dysfunctional Cats"
+    assert cdf_manager.global_attributes["PI_name"] == "Ana Manica"
+    assert cdf_manager.global_attributes["PI_affiliation"] == "LASP, CU"
+    assert cdf_manager.global_attributes["Data_version"] == 2
+    assert cdf_manager.global_attributes["DOI"] == "test"
 
-    assert cdf_manager.global_attributes["Descriptor"] == "MAG>Magnetometer"
+    # Testing that everything loaded into the global attrs is present in
+    #   the global attrs schema
+    for attr in cdf_manager.global_attributes.keys():
+        assert attr in cdf_manager.global_attribute_schema.keys()
 
-    mag_l1a_global_attrs = cdf_manager.get_global_attributes("imap_mag_l1a_norm-raw")
 
+def test_get_global_attributes():
+    """
+    Test function that covers:
+        get_global_attributes
+    """
+    # Initialize CdfAttributeManager object which loads in default info
+    cdf_manager = CdfAttributeManager(Path(__file__).parent.parent / "config")
+
+    # Change filepath to load test global attributes
+    cdf_manager.source_dir = Path(__file__).parent.parent / "tests"
+    cdf_manager.load_global_attributes("imap_default_global_test_cdf_attrs.yaml")
+    cdf_manager.load_global_attributes("imap_test_global.yaml")
+
+    # Loading in instrument specific attributes
+    test_get_global_attrs = cdf_manager.get_global_attributes("imap_test_T1_test")
+
+    # Testing information previously loaded into global attributes (first if statement)
+    assert test_get_global_attrs["Project"] == "STP>Solar-Terrestrial Physics"
     assert (
-        mag_l1a_global_attrs["Mission_group"]
+        test_get_global_attrs["Source_name"]
         == "IMAP>Interstellar Mapping and Acceleration Probe"
     )
-    assert mag_l1a_global_attrs["Descriptor"] == "MAG>Magnetometer"
-    assert mag_l1a_global_attrs["Logical_source"] == "imap_mag_l1a_norm-raw"
+    assert test_get_global_attrs["Mission_group"] == "Dysfunctional Cats"
+    # Testing instrument specific global attributes (first elif statement)
+    assert test_get_global_attrs["Descriptor"] == "TEST>Testinstrument"
+    assert test_get_global_attrs["Data_type"] == "T1_test-one>Test-1 test one"
+    assert test_get_global_attrs["Logical_source"] == "imap_test_T1_test"
+    assert (
+        test_get_global_attrs["Logical_source_description"]
+        == "IMAP Mission TEST one document Level-T1."
+    )
+    # Not given, and not required information
+    assert test_get_global_attrs["Data_level"] is None
+    with pytest.raises(KeyError):
+        assert test_get_global_attrs["bad_name"] == "false info"
```

Comment:
You should also test that a global attribute in the file but not in the schema isn't passed along. So for example, add a new attribute `not_passed_along` inside imap_test_global.yaml and then assert that it isn't present in `test_get_global_attrs`